src/coreComponents/linearAlgebra/interfaces/hypre/HypreVector.cpp

```diff
@@ -116,216 +87,80 @@ HypreVector::~HypreVector()
-}
-
-void HypreVector::set( arraySlice1d< globalIndex const > const & globalIndices,
-                       arraySlice1d< real64 const > const & values )
-{
-  GEOSX_LAI_ASSERT( !closed() );
-  GEOSX_LAI_ASSERT_GE( *std::min_element( globalIndices.dataIfContiguous(),
-                                          globalIndices.dataIfContiguous() + globalIndices.size() ), ilower() );
-  GEOSX_LAI_ASSERT_GT( iupper(), *std::max_element( globalIndices.dataIfContiguous(),
-                                                    globalIndices.dataIfContiguous() + globalIndices.size() ) );
-
-  GEOSX_LAI_CHECK_ERROR( HYPRE_IJVectorSetValues( m_ij_vector,
-                                                  LvArray::integerConversion< HYPRE_Int >( values.size() ),
-                                                  hypre::toHypreBigInt( globalIndices.dataIfContiguous() ),
-                                                  values.dataIfContiguous() ) );
-}
-
-void HypreVector::add( arraySlice1d< globalIndex const > const & globalIndices,
-                       arraySlice1d< real64 const > const & values )
-{
-  GEOSX_LAI_ASSERT( !closed() );
-  GEOSX_LAI_CHECK_ERROR( HYPRE_IJVectorAddToValues( m_ij_vector,
-                                                    LvArray::integerConversion< HYPRE_Int >( values.size() ),
-                                                    hypre::toHypreBigInt( globalIndices.dataIfContiguous() ),
-                                                    values.dataIfContiguous() ) );
+  return m_vec != nullptr;
 }
 
 void HypreVector::set( real64 value )
 {
   GEOSX_LAI_ASSERT( ready() );
-  GEOSX_LAI_CHECK_ERROR( HYPRE_ParVectorSetConstantValues( m_par_vector, value ) );
-}
-
-void HypreVector::zero()
-{
-  GEOSX_LAI_ASSERT( ready() );
-  GEOSX_LAI_CHECK_ERROR( hypre_IJVectorZeroValues( m_ij_vector ) );
+  GEOSX_LAI_CHECK_ERROR( HYPRE_ParVectorSetConstantValues( m_vec, value ) );
+  touch();
 }
 
 void HypreVector::rand( unsigned const seed )
 {
   GEOSX_LAI_ASSERT( ready() );
-  GEOSX_LAI_CHECK_ERROR( HYPRE_ParVectorSetRandomValues( m_par_vector, seed ) );
-}
-
-void HypreVector::open()
-{
-  GEOSX_LAI_ASSERT( ready() );
-  GEOSX_LAI_CHECK_ERROR( HYPRE_IJVectorInitialize( m_ij_vector ) );
-  m_closed = false;
+  GEOSX_LAI_CHECK_ERROR( HYPRE_ParVectorSetRandomValues( m_vec, seed ) );
+  touch();
 }
 
 void HypreVector::close()
 {
   GEOSX_LAI_ASSERT( !closed() );
-  GEOSX_LAI_CHECK_ERROR( HYPRE_IJVectorAssemble( m_ij_vector ) );
+  m_values.move( hypre::memorySpace, false );
   m_closed = true;
 }
 
+void HypreVector::touch()
+{
+  GEOSX_LAI_ASSERT( ready() );
+  m_values.registerTouch( hypre::memorySpace );
+}
+
 void HypreVector::scale( real64 const scalingFactor )
 {
   GEOSX_LAI_ASSERT( ready() );
   if( !isEqual( scalingFactor, 1.0 ) )
   {
-    GEOSX_LAI_CHECK_ERROR( HYPRE_ParVectorScale( scalingFactor, m_par_vector ) );
+    GEOSX_LAI_CHECK_ERROR( HYPRE_ParVectorScale( scalingFactor, m_vec ) );
```

Comment:
wouldn't this be easier to do on our own?

Reply:
Yeah, it's a super trivial kernel to write. In general I try to only write a custom kernel when a hypre method doesn't exist or doesn't work properly on device, the idea being that hypre folks might spend a bit more time optimizing it (e.g. CUDA block sizes). For this simple kernel it's unlikely to matter.